src/ShadyCSS/custom-style.js

```diff
@@ -0,0 +1,146 @@
+/**
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+/*
+Wrapper over <style> elements to co-operate with ShadyCSS
+
+Example:
+<shady-style>
+  <style>
+  ...
+  </style>
+</shady-style>
+*/
+
+'use strict';
+
+let ShadyCSS = window.ShadyCSS;
+
+let enqueued = false;
+
+let customStyles = [];
+
+let hookFn = null;
+
+function enqueueDocumentValidation() {
+  if (enqueued) {
+    return;
+  }
+  enqueued = true;
+  if (window.HTMLImports) {
```

Comment:
Let's add a comment about the strategy here and acknowledge that there can be FOUC